packages/utils/src/activity.js

```diff
@@ -44,12 +32,30 @@ export function getAncestors(activities, activity) {
   return [...ancestors, parent];
 }
 
-export function toTreeFormat(activities, targetLevels, parentId = null, level = 1) {
-  return getOutlineChildren(activities, parentId).map(activity => ({
+export function getOutlineChildren(activities, parentId, schema) {
+  const children = getChildren(activities, parentId);
+  if (!parentId || !children.length) return children;
+  const parentType = find(activities, { id: parentId }).type;
+  const types = schema.getLevel(parentType).subLevels;
+  return filter(children, it => types.includes(it.type));
+}
+
+export function getOutlineChildrenFilterFn(schema) {
```

Comment:
Consider not using the same function for resolving children in the hierarchy and filtering out specific types. That way you don't have to pass the filter function always and you don't have to have a more specific signature expecting parentId. To recap, instead of:
`return filterNodesFn(activities, parentId)`
you can do
`const parentActivities = filter(activities, { parentId })`
and then apply filtering. Instead of getOutlineChildrenFilterFn
you can pass isOutlineAcitivty which is already exposed within the schema config package.